adapters/repos/db/index.go

```diff
@@ -391,25 +389,24 @@ func (i *Index) initAndStoreShards(ctx context.Context, shardState *sharding.Sta
 
 // used to init/create shard in different moments of index's lifecycle, therefore it needs to be called
 // within shardCreateLocks to prevent parallel create/init of the same shard
-func (i *Index) initAndStoreShard(ctx context.Context, shardName string, class *models.Class,
-	promMetrics *monitoring.PrometheusMetrics,
-) error {
-	if i.Config.DisableLazyLoadShards {
+func (i *Index) initShard(ctx context.Context, shardName string, class *models.Class,
```

Comment:
I think it would be nice to update the godoc on this and related `init...` function to explain when to use vs the other and what they entail compared to the other ones. The code still feels confusing when reading it trying to follow along what is expected to do what.